### ISSUE TITLE
order execution issue when market Open/Close

### ISSUE THREAD

**Issue by mnf1980** [mnf1980/SIM#38]
The stock open at 4.31 but in the SIM it executed at 4.10.
the order was to buy 100 share @ limit up at 4.96.
the same also for the CHAEIN.

![35](https://user-images.githubusercontent.com/84329389/220835355-fe6a0d0a-475d-4b0e-8cd9-9313399a035e.png)

**Comment by Samiaatieh** [comment]
best ask 4.1 better than limit price 4.96 that why the order is completed with 4.1

If you buy and the price is more than ASK it will be as market value 
if you sell and the Price  is  less than Bid it will be as market value 

The market working like this, please check with your team and let me know

**Comment by mnf1980** [comment]
you are right, but it is not the case here.
the order is a limit order to buy at 4.96 and placed in **pre-open** session. the correct behavior is to execute the order when market open at open price 4.31. 
the case you mention is correct during the market session.

**Comment by Samiaatieh** [comment]
Why 4.31 ??

**Comment by mnf1980** [comment]
this is the open price that day. 
Market open at 4.31 and all orders placed in the pre-open session must executed at that price.

**Comment by mnf1980** [comment]
you can also check the picture attached above (upper part of the pic) that shows the open price at ADX market watch.

**Comment by mnf1980** [comment]
we may have the same issue at market close, please see the pic:
- order placed to buy CHAESH @ 7.96 in **pre-close session**
- at 2:55 the system executes the trade at best offer (**not the closing price**)
- correct behavior is to **close at closing price** 

![38](https://user-images.githubusercontent.com/84329389/223092348-958bdb8d-9f44-4514-aa31-7980bda70d2b.png)

**Comment by Samiaatieh** [comment]
Can you check it today

**Comment by Samiaatieh** [comment]
Pending:  ETF should be 2:45 to 2:55 order should not be completed.

**Comment by mnf1980** [comment]
yes, right, order can be placed but not executed between 2:45 to 2:55.

this is what happened today please check.
![45](https://user-images.githubusercontent.com/84329389/233978698-bb4f3b85-cd75-4fb1-88d0-0693e2d26976.png)

**Comment by mnf1980** [comment]
i found that the issue is also when market open,
- i put order to buy ADCB at 10.06 in the pre-open session
- ADCB open at 8.66
- order executed at 7.88


![46](https://user-images.githubusercontent.com/84329389/234188140-8c0f3638-3a7e-4999-a4f8-61b5f0faac04.png)

**Comment by Samiaatieh** [comment]
Dear Manaf,

I'm doing testing so  I changed the market status to Pre closed to check
the ETF that's why you had this
Sorry I should inform you before you do testing

Regards,

Samia Atieh
Product Manager

Tel.  +971-4-433-4386    Mob +971-55-8694948   Skype: susjamil
Email: ***@***.***
www.ZagTrader.com
